packages/common/container-definitions/src/loader.ts

```diff
@@ -102,23 +102,134 @@ export interface ICodeAllowList {
 }
 
 /**
- * Events emitted by the Container "upwards" to the Loader and Host
+ * Events emitted by the {@link IContainer} "upwards" to the Loader and Host.
  */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IContainerEvents extends IEvent {
+    /**
+     * Emitted when the readonly state of the container changes.
+     *
+     * @remarks Listener parameters:
+     *
+     * - `readonly`: Whether or not the container is now in a readonly state.
+     *
+     * @see {@link IContainer.readOnlyInfo}
+     */
     (event: "readonly", listener: (readonly: boolean) => void): void;
+
+    /**
+     * Emitted when the {@link IContainer} completes connecting to the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.connect}
+     */
     (event: "connected", listener: (clientId: string) => void);
+
+    /**
+     * TODO
```

Comment:
Fires when new container code details have been proposed, prior to acceptance.

codeDetails: the details that have been proposed

proposal: Not recommended for use

Reply:
Is the intention to remove the "proposal" param then?

Reply:
And is there any further elaboration we can offer on code details proposal? This is an area in which I am still personally hazy. If we could point to other docs or something, that might be nice.

Reply:
Re: removing - would probably be a good idea!  I haven't really considered it though :)

Re: more details - Yes, I've been working on our versioning story which relates to this.  We've shifted our guidance in this area quite a few times though, so I want to be somewhat cautious about how we communicate here to avoid giving customers whiplash.  I consider part of my responsibility in that project is to explain code proposal/version.